### PR TITLE
Backend skeleton

### LINKED ISSUE #3: setup user Schema

### LINKED ISSUE #4: setup trips schema

### LINKED ISSUE #5: set up tags schema

### LINKED ISSUE #7: set up votes schema

### LINKED ISSUE #62: setup destinations schema

### LINKED ISSUE #64: setup users_trips schema

### LINKED ISSUE #65: setup friends schema

### PR DESCRIPTION
**Pull request by leranf**
#### What's this PR do?

Creates schemas for:
- destinations closes #62, #63 (accidentally added duplicate)
- tags closes #5 
- trips closes #4 
- userTrips closes #64
- users closes #3 
- friends closes #65 
- votes closes #7  
  #### Any background context you want to provide?
  n/a
  #### What are the relevant tickets/issues?
  see above
  models' controller functions
  #### Where should the reviewer start?
  server/
  #### Are there tests written yet? How should this be manually tested?
  no, start up the server and check if tables were created correctly
  #### Screenshots (if appropriate)
  #### Questions:
  - Does this add new dependencies? 'require'
  - Does our documentation need to be updated? no

### CODE REVIEW

**Comment by leranf** [comment]
@abuddiga @boyajay @xuejinglu please review

File: server/destinations/destinations.js [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
 const Destination = db.define('destinations', {
   startDate: Sequelize.DATE,
   endDate: Sequelize.DATE,
   location: Sequelize.STRING,
-  tripId: Sequelize.INTEGER,
 });
 
+// creates tripId column in destinations table
+Destination.belongsTo(Trip);
+
```
Comment (abuddiga) [inline server/destinations/destinations.js:13]: should also include `Trip.hasMany(Destination);`

File: server/destinations/destinations.js [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
 Destination.sync();
```
Comment (abuddiga) [inline server/destinations/destinations.js:19]: should include Trip.sync();

File: server/tags/tags.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const db = require('../config/db');
+const Sequelize = require('sequelize');
+const User = require('../users/user');
```
Comment (abuddiga) [inline server/tags/tags.js]: should be users/users

File: server/tags/tags.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+const Event = require('../events/events');
+
+const Tag = db.define('tags', {
+  name: Sequelize.STRING,
+});
+
+// creates eventId column in tags table
+Tag.belongsTo(Event);
+
```
Comment (abuddiga) [inline server/tags/tags.js:12]: create same hasMany relation, and sync (like above)

File: server/trips/trips.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+});
+
+const UserTrip = db.define('userTrips', {
+  viewed: Sequelize.BOOLEAN,
+});
+
+// creates tripId column in userTrips join table
+// creates userId column in userTrips join table
+User.belongsToMany(Trip, { through: UserTrip });
+Trip.belongsToMany(User, { through: UserTrip });
```
Comment (boyajay) [inline server/trips/trips.js:17]: add 'Trip hasMany Users', 'Users hasMany Trips'

File: server/users/users.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  name: Sequelize.STRING,
+  picUrl: Sequelize.STRING,
+});
+
+const UserFriend = db.define('friends', {});
+
+// defines a self reference to user
+// creates userId column in friends join table
+// creates friendId column in friends join table
+User.belongsToMany(User, { as: 'friend', through: UserFriend });
```
Comment (abuddiga) [inline server/users/users.js]: as: 'Friends"

File: server/users/users.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+User.sync();
+UserFriend.sync();
```
Comment (abuddiga) [inline server/users/users.js:18]: sync UserFriend first, before User

File: server/trips/trips.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+Trip.sync();
+User.sync();
+UserTrip.sync();
```
Comment (abuddiga) [inline server/trips/trips.js]: sync UserTrip first

File: server/votes/votes.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+const User = require('../users/users');
+const Event = require('../events/events');
+
+const Vote = db.define('votes', {});
+
+// creates userId column in votes table
+// creates eventId column in votes table
+Vote.belongsTo(User);
+Vote.belongsTo(Event);
+
```
Comment (abuddiga) [inline server/votes/votes.js:12]: set up hasMany relations as well, and sync

**Pull request merged**